File: Stack/python/Stack.py
from Node import Node

class Stack:
    def __init__(self):
        self.last = None

    def is_empty(self):
        return self.last is None

    def push(self, value):
        node = Node(value)
        if self.last:
            temp = self.last
            self.last = node
            self.last.next = temp
        else:
            self.last = node
    
    def pop(self):
        if self.is_empty() == True:
            raise IndexError('Stack is empty')
        temp = self.last
        self.last = self.last.next
        return temp.value

    def peek(self):
        return self.last.value


        

File: Stack/java/Stack.java
public class Stack<T>{
    private Node<T> head;


    public Node<T> getHead(){
        return this.head;
    }

    Stack(){
        this.head = null;
    }

    public boolean isEmpty(){
        return this.head == null;
    }

    public void push(T value){
        Node<T> newNode = new Node<>(value);
        if(this.isEmpty()){
            this.head = newNode;
        }else{
            newNode.next = this.head;
            this.head = newNode;
        }
    }
    public T pop(){
        if(this.isEmpty()){
            System.out.println("Stack is empty");
            return null; 
        }
        else{
            T temp = this.head.value;
            this.head = this.head.next;
            return temp;
        }
    }

    public T peek(){
        if(this.isEmpty()){
            return null;
        }else{
            return this.head.value;
            }
        }
    }

